app/gui/src/project-view/stores/project/index.ts

```diff
@@ -250,7 +251,12 @@ export const { provideFn: provideProjectStore, injectFn: useProjectStore } = cre
         { immediate: true, flush: 'post' },
       )
 
-      return computed(() => parseVisualizationData(visualizationDataRegistry.getRawData(id)))
+      return {
+        data: computed(() =>
+          parseVisualizationData(visualizationDataRegistry.getRawData(id.value)),
+        ),
+        reconnect: () => (id.value = newId()),
```

Comment:
Could we also automatically regenerate ID always when preprocessor (`expression` or `visualizationModule`) changed? I think this should be a "general" rule, not only for visualization displayed under graph.

It may be a bit tricky to avoid recursive `watch` run: my idea is to make current `id` ref to be a tuple of (`id`, config); the current watch just watches this tuple, we add additional watch of `configuration` which sets new tuple re-using or regenerating id, and `reconnect` just regenerates id while keeping config. But maybe it may be even simpler.